internal/app/fipcontroller/hcloud.go

```diff
@@ -35,28 +35,92 @@ func (controller *Controller) floatingIP(ctx context.Context, ipAddress string)
 }
 
 /*
- * Search and return the hcloud Server object for a given IP address.
- *  The IP Address can be a public IPv4, IPv6 address or a private address attached to any private network interface
+ * Search and return the hcloud Server objects for a given list of IP addresses.
+ *  The IP Addresses can be public IPv4, IPv6 addresses or private addresses attached to any private network interface
  */
-func (controller *Controller) server(ctx context.Context, ip net.IP) (server *hcloud.Server, err error) {
+func (controller *Controller) servers(ctx context.Context, ips []net.IP) (serverList []*hcloud.Server, err error) {
+	// Fetch all hetzner servers
 	servers, err := controller.HetznerClient.Server.All(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("could not fetch servers: %v", err)
 	}
 	controller.Logger.Debugf("Fetched %d servers", len(servers))
 
+	for _, ip := range ips {
+		// check if a server was found for the given ip, if not throw error
+		server := controller.searchServerForIP(servers, ip)
+		if server == nil {
+			return nil, fmt.Errorf("Could not find an a server for ip '%s'", ip)
+		}
+		serverList = append(serverList, server)
+	}
+	return serverList, nil
+}
+
+/*
+ * Search for a hetzner Server that has the give ip in any of its networks
+ */
+func (controller *Controller) searchServerForIP(servers []*hcloud.Server, ip net.IP) *hcloud.Server {
 	for _, server := range servers {
 		// IP must not be a floating IP, but might be private or public depending on the cluster configuration
 		if server.PublicNet.IPv4.IP.Equal(ip) || server.PublicNet.IPv6.IP.Equal(ip) {
 			controller.Logger.Debugf("Found matching public IP on server '%s'", server.Name)
-			return server, nil
+			return server
+		}
+
+		privateNet := controller.searchPrivateNet(server.PrivateNet, ip)
+		if privateNet {
```

Comment:
Any specific reason the privateNet variable is assigned outside of the if-context?